server/internal/server/completions.go

```diff
@@ -57,22 +57,24 @@ func (s *S) CreateChatCompletion(
 		return
 	}
 
-	dest, err := s.engineGetter.GetEngineForModel(req.Context(), createReq.Model)
-	if err != nil {
-		http.Error(w, fmt.Sprintf("Failed to find pod to route the request: %s", err), http.StatusInternalServerError)
-		return
-	}
-
 	// Check if the specified model is available
-	ctx := auth.CarryMetadataFromHTTPHeader(req.Context(), req.Header)
-	if _, err := s.modelClient.GetModel(ctx, &mv1.GetModelRequest{
-		Id: createReq.Model,
-	}); err != nil {
-		if status.Code(err) == codes.NotFound {
-			http.Error(w, fmt.Sprintf("Model not found: %s", createReq.Model), http.StatusBadRequest)
+	if s.enableAuth {
+		ctx := auth.CarryMetadataFromHTTPHeader(req.Context(), req.Header)
+		if _, err := s.modelClient.GetModel(ctx, &mv1.GetModelRequest{
+			Id: createReq.Model,
+		}); err != nil {
+			if status.Code(err) == codes.NotFound {
+				http.Error(w, fmt.Sprintf("Model not found: %s", createReq.Model), http.StatusBadRequest)
+				return
+			}
+			http.Error(w, fmt.Sprintf("Failed to get model: %s", err), http.StatusInternalServerError)
 			return
 		}
-		http.Error(w, fmt.Sprintf("Failed to get model: %s", err), http.StatusInternalServerError)
+	}
+
+	dest, err := s.engineGetter.GetEngineForModel(req.Context(), createReq.Model)
```

Comment:
Since it takes a long time to pull model to the engine, it is better to check if a user is allow to access the model before pulling it.